cache/in-memory/src/permission.rs

```diff
@@ -361,7 +361,15 @@ impl<'a> InMemoryCachePermissions<'a> {
         let calculator =
             PermissionCalculator::new(guild_id, user_id, everyone, assigned.as_slice());
 
-        Ok(calculator.in_channel(channel.kind, overwrites.as_slice()))
+        let permissions = calculator.in_channel(channel.kind, overwrites.as_slice());
+
+        if !permissions.contains(Permissions::ADMINISTRATOR)
+            && self.is_communication_disabled(user_id, guild_id)
+        {
+            return Ok(Permissions::VIEW_CHANNEL | Permissions::READ_MESSAGE_HISTORY);
```

Comment:
This will be wrong if you are timed out but request permissions on a channel where you don't have either permission. Also pretty sure it doesn't disable permissions like manage roles and similar guild level permissions

Reply:
thats a good point! and i got those from the discord docs

> Timed out members will temporarily lose all permissions except VIEW_CHANNEL and READ_MESSAGE_HISTORY. Owners and admin users with ADMINISTRATOR permissions are exempt.

but maybe its just implicit, needs some testing i think